tests/kafkatest/tests/quota_test.py

```diff
@@ -0,0 +1,164 @@
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ducktape.tests.test import Test
+from ducktape.utils.util import wait_until
+
+from kafkatest.services.zookeeper import ZookeeperService
+from kafkatest.services.kafka import KafkaService
+from kafkatest.services.verifiable_producer import VerifiableProducer
+from kafkatest.services.performance import ProducerPerformanceService
+from kafkatest.services.console_consumer import ConsoleConsumer, is_int
+
+import random
+import signal
+import time
+
+class QuotaTest(Test):
+    """
+    These tests verify that quota provides expected functionality -- they run
+    producer, broker, and consumer with different clientId and quota configuration and
+    check that the observed throughput is close to the value we expect.
+    """
+
+    def __init__(self, test_context):
+        """:type test_context: ducktape.tests.test.TestContext"""
+        super(QuotaTest, self).__init__(test_context=test_context)
+
+        self.topic = "test_topic"
+        self.logger.info("use topic " + self.topic)
+
+        # quota related parameters
+        self.quota_config = {"quota_producer_default": 2500000,
+                             "quota_consumer_default": 2000000,
+                             "quota_producer_bytes_per_second_overrides": "overridden_id=3750000",
+                             "quota_consumer_bytes_per_second_overrides": "overridden_id=3000000"}
+        self.maximum_deviation_percentage = 100.0
+        self.num_records = 100000
+        self.record_size = 3000
+
+        self.zk = ZookeeperService(test_context, num_nodes=1)
+        self.kafka = KafkaService(test_context, num_nodes=1, zk=self.zk,
+                                  topics={self.topic: {"partitions": 6, "replication-factor": 1, "min.insync.replicas": 1}},
+                                  quota_config=self.quota_config)
+        self.num_producers = 1
+        self.num_consumers = 2
+
+    def setUp(self):
+        self.zk.start()
+        self.kafka.start()
+
+    def min_cluster_size(self):
+        """Override this since we're adding services outside of the constructor"""
+        return super(QuotaTest, self).min_cluster_size() + self.num_producers + self.num_consumers
+
+    def run_clients(self, producer_id, producer_num, consumer_id, consumer_num):
```

Comment:
@granders I haven't seen any use of this feature in kafkatests. Does this allow developer to specify test names, e.g. test_overridden_quota, so that we can get an idea of what the test is in the report?